connectors/sources/servicenow.py

```diff
@@ -0,0 +1,272 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""ServiceNow source module responsible to fetch documents from ServiceNow."""
+import json
+
+import aiohttp
+from aiohttp.client_exceptions import ServerDisconnectedError
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource
+from connectors.utils import CancellableSleeps, iso_utc
+
+RETRY_INTERVAL = 2
+FETCH_SIZE = 1000
+MAX_CONCURRENCY_SUPPORT = 15
+
+ENDPOINTS = {
+    "TABLE": "/api/now/table/{table}",
+}
+
+
+class InvalidResponse(Exception):
+    """Raise when retrieving invalid response..
+
+    Args:
+        Exception (InvalidResponse): Invalid response exception.
+    """
+
+    pass
+
+
+class ServiceNowDataSource(BaseDataSource):
+    """ServiceNow"""
+
+    name = "ServiceNow"
+    service_type = "servicenow"
+
+    def __init__(self, configuration):
+        """Setup the connection to the ServiceNow instance.
+
+        Args:
+            configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
+        """
+
+        super().__init__(configuration=configuration)
+        self.session = None
+        self.sleeps = CancellableSleeps()
+        self.retry_count = self.configuration["retry_count"]
+        self.concurrent_downloads = self.configuration["concurrent_downloads"]
+
+    def tweak_bulk_options(self, options):
+        """Tweak bulk options as per concurrent downloads support by ServiceNow
+
+        Args:
+            options (dict): Config bulker options.
+        """
+
+        options["concurrent_downloads"] = self.concurrent_downloads
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for ServiceNow.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "url": {
```

Comment:
referring to [this](https://github.com/elastic/connectors-python/pull/537#discussion_r1129641667) discussion. Could you please remove remove "non-default" fields?